### PR TITLE
Use `color` for color.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
Color functionality is now provided by the `color` crate.

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
Some of these will be resolved in `color` and some here in this crate:

* [ ] `PartialEq`
* [ ] `PartialOrd` (needed by `ColorStop` at least)
* [ ] `Hash`
* [ ] Add `legacy_color` feature (enabled by default for now) that feature gates the existing `peniko::Color`
* [ ] Do the work on `Gradient`
* [ ] Better solution for disabling some clippy lints for `allow` attributes
* [ ] `serde` support in `color`
* [ ] Decide what type(s) to export

**Comment by waywardmonkeys** [comment]
I've done more work here updating most of the gradient code, adding a feature to disable the legacy color code, and adding some more comments.

There are some `FIXME(color)` comments in the code to help point to things that need resolution in conjunction with the list in a comment above.

File: src/brush.rs [hunk 1]
```diff
@@ -37,7 +48,8 @@ impl From<Image> for Brush {
 
 impl Default for Brush {
     fn default() -> Self {
-        Self::Solid(Color::default())
+        // FIXME(color): Decide what to do about color and defaults
+        Self::Solid(color::palette::css::TRANSPARENT)
```
Comment (tomcur) [inline src/brush.rs:52]: Current Peniko defaults to transparent. Opaque black can be considered: it is visible, and has a natural representation in the color spaces we include in `color`. The other special color is white, though the physical color it represents depends on the color space.

Comment (waywardmonkeys) [reply src/brush.rs:52]: There was a previous PR with discussion about it: https://github.com/linebender/peniko/pull/31 and that links to more discussion in Xilem.

We'll also have to decide whether to solve it here in the `Brush::default()` only or also in `color` or both.

Comment (waywardmonkeys) [reply src/brush.rs:52]: Also, with https://github.com/linebender/color/issues/63 in place, this would be shorter …

**Comment by waywardmonkeys** [comment]
I think I will remove the legacy code instead, and export a type alias `Color` that is the same as `AlphaColor<Srgb>` as that would simplify updating Vello.

**Comment by tomcur** [comment]
(Just a note: with color becoming 32-bit per channel, maybe `peniko::Image::alpha` should become `f32`.)

https://github.com/linebender/peniko/blob/d825bdf5039d592889a0b2816a246cd5fa22bae8/src/image.rs#L37-L50

File: src/brush.rs (hunk 1 above)
Comment (DJMcNab) [reply src/brush.rs]: I don't think we should solve this in Color.

File: src/lib.rs [hunk 2]
```diff
@@ -30,20 +31,24 @@
...
 pub use blob::{Blob, WeakBlob};
 pub use brush::{Brush, BrushRef, Extend};
-pub use color::Color;
 pub use font::Font;
 pub use gradient::{ColorStop, ColorStops, ColorStopsSource, Gradient, GradientKind};
 pub use image::{Format, Image};
 pub use style::{Fill, Style, StyleRef};
+
+/// FIXME(color): Write something
+pub type Color = color::AlphaColor<color::Srgb>;
```
Comment (DJMcNab) [inline src/lib.rs:54]: I'd be tempted to remove this alias entirely.

File: Cargo.toml [hunk 3]
```diff
@@ -69,8 +74,10 @@ rust.unused_macro_rules = "warn"
 rust.unused_qualifications = "warn"
 rust.variant_size_differences = "warn"
 
-clippy.allow_attributes = "warn"
-clippy.allow_attributes_without_reason = "warn"
+# FIXME(color): I'm lazy about fixing this yet.
+clippy.allow_attributes = "allow"
```
Comment (DJMcNab) [inline Cargo.toml:78]: This should be a crate level exception, I think.

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ You can find its changes [documented below](#020-2024-09-19).
 
 ## [Unreleased]
 
-This release has an [MSRV] of 1.70.
+This release has an [MSRV] of 1.82.
```
Comment (DJMcNab) [inline CHANGELOG.md:17]: Something more here?

File: Cargo.toml (hunk 3 above)
Comment (waywardmonkeys) [reply Cargo.toml:78]: I'd like to see the MSRV bump to 1.82 happen independently in preparation of this landing and so this sort of change won't be needed here.

File: CHANGELOG.md (hunk 4 above)
Comment (waywardmonkeys) [reply CHANGELOG.md:17]: A changelog entry has now been written.

File: Cargo.toml (hunk 3 above)
Comment (waywardmonkeys) [reply Cargo.toml]: Between #65 and here, I've resolved the MSRV stuff.

**Comment by waywardmonkeys** [comment]
> (Just a note: with color becoming 32-bit per channel, maybe `peniko::Image::alpha` should become `f32`.)

I did this in #65 as it let me simplify some of the MSRV issues.

File: CHANGELOG.md [hunk 5]
```diff
@@ -13,11 +13,37 @@ You can find its changes [documented below](#020-2024-09-19).
...
+- `Brush`, `BrushRef`, and `ColorStop` can be constructed from a variety of color types,
+  although, for now, `Brush` and `BrushRef` convert this internally into an unclipped
+  `AlphaColor<Srgb>`.
+- The `color` crate is re-exported as `peniko::color`, so access to functionality from
+  there is easy.
+- The various pre-defined color constants are no longer available like
+  `peniko::Color::YELLOW`. Instead, use the CSS palette provided within `color` like
+  `peniko::color::palette::css::YELLOW`.
+- Similarly, parsing a color string is now provided by the `color` crate.
+- The MSRV is now 1.82 as that is the MSRV of `color.
```
Comment (xStrom) [inline CHANGELOG.md:43]:
```suggestion
- The MSRV is now 1.82 as that is the MSRV of `color`.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -13,11 +13,37 @@ You can find its changes [documented below](#020-2024-09-19).
...
 
 ### Changed
 
 - `Image` now stores the alpha as an `f32` ([#65][] by [@waywardmonkeys][])
+- Use `color` crate. See below for details. ([#63][] by [@waywardmonkeys][])
+
+### Color Changes
+
+The old code behind `peniko::Color` has been removed and color functionality is now provided by
+the [`color`] crate.
```
Comment (xStrom) [inline CHANGELOG.md:26]: In Markdown files let's keep to our [policy of one line per sentence](https://linebender.org/wiki/formatting-scheme/#markdown).

```suggestion
The old code behind `peniko::Color` has been removed and color functionality is now provided by the [`color`] crate.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -13,11 +13,37 @@ You can find its changes [documented below](#020-2024-09-19).
...
+
+This leads to a number of breaking changes:
+
+- `peniko::Color` is now a type alias for `AlphaColor<Srgb>` from the `color` crate.
+- `AlphaColor` does not, at this time, impl `Default`, `PartialEq`, `PartialOrd`, or `Hash`.
+- `Brush` and `BrushRef` no longer impl `PartialEq`.
+- `ColorStop` no longer impls `Default` or `PartialOrd`.
+- `Brush`, `BrushRef`, and `ColorStop` can be constructed from a variety of color types,
+  although, for now, `Brush` and `BrushRef` convert this internally into an unclipped
+  `AlphaColor<Srgb>`.
```
Comment (xStrom) [inline CHANGELOG.md:36]: One line.

```suggestion
- `Brush`, `BrushRef`, and `ColorStop` can be constructed from a variety of color types, although, for now, `Brush` and `BrushRef` convert this internally into an unclipped `AlphaColor<Srgb>`.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -13,11 +13,37 @@ You can find its changes [documented below](#020-2024-09-19).
...
+- The `color` crate is re-exported as `peniko::color`, so access to functionality from
+  there is easy.
```
Comment (xStrom) [inline CHANGELOG.md:38]: One line.

```suggestion
- The `color` crate is re-exported as `peniko::color`, so access to functionality from there is easy.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -13,11 +13,37 @@ You can find its changes [documented below](#020-2024-09-19).
...
+- The various pre-defined color constants are no longer available like
+  `peniko::Color::YELLOW`. Instead, use the CSS palette provided within `color` like
+  `peniko::color::palette::css::YELLOW`.
```
Comment (xStrom) [inline CHANGELOG.md:41]: One line.

```suggestion
- The various pre-defined color constants are no longer available like `peniko::Color::YELLOW`.
  Instead, use the CSS palette provided within `color` like `peniko::color::palette::css::YELLOW`.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -13,11 +13,37 @@ You can find its changes [documented below](#020-2024-09-19).
...
+- Similarly, parsing a color string is now provided by the `color` crate.
+- The MSRV is now 1.82 as that is the MSRV of `color.
+
+This is the first step towards providing better support for richer color functionality
+throughout the Linebender stack.
```
Comment (xStrom) [inline CHANGELOG.md:46]: One line.

```suggestion
This is the first step towards providing better support for richer color functionality throughout the Linebender stack.
```

**Review by xStrom — COMMENTED** [review]
Some initial thoughts on the changelog. 😈

File: CHANGELOG.md [hunk 11]
```diff
@@ -13,11 +13,30 @@ You can find its changes [documented below](#020-2024-09-19).
...
+
+This leads to a number of breaking changes:
+
+- `peniko::Color` is now a type alias for `AlphaColor<Srgb>` from the `color` crate.
+- `AlphaColor` does not, at this time, impl `Default`, `PartialEq`, `PartialOrd`, or `Hash`.
+- `Brush` and `BrushRef` no longer impl `PartialEq`.
+- `ColorStop` no longer impls `Default` or `PartialOrd`.
+- `Brush`, `BrushRef`, and `ColorStop` can be constructed from a variety of color types, although, for now, `Brush` and `BrushRef` convert this internally into an unclipped `AlphaColor<Srgb>`.
+- The `color` crate is re-exported as `peniko::color`, so access to functionality from there is easy.
+- The various pre-defined color constants are no longer available like `peniko::Color::YELLOW`. Instead, use the CSS palette provided within `color` like `peniko::color::palette::css::YELLOW`.
```
Comment (xStrom) [inline CHANGELOG.md:35]: This one is two sentences, so I think the following split is correct.

```suggestion
- The various pre-defined color constants are no longer available like `peniko::Color::YELLOW`.
  Instead, use the CSS palette provided within `color` like `peniko::color::palette::css::YELLOW`.
```

**Review by xStrom — APPROVED** [review]
I have not tried running the code with Vello or anything like that, but the code changes look fine to me.

File: CHANGELOG.md [hunk 12]
```diff
@@ -13,11 +13,31 @@ You can find its changes [documented below](#020-2024-09-19).
...
+- `peniko::Color` is now a type alias for `AlphaColor<Srgb>` from the `color` crate.
+- `AlphaColor` does not, at this time, impl `Default`, `PartialEq`, `PartialOrd`, or `Hash`.
+- `Brush` and `BrushRef` no longer impl `PartialEq`.
+- `ColorStop` no longer impls `Default` or `PartialOrd`.
+- `Brush`, `BrushRef`, and `ColorStop` can be constructed from a variety of color types, although, for now, `Brush` and `BrushRef` convert this internally into an unclipped `AlphaColor<Srgb>`.
+- The `color` crate is re-exported as `peniko::color`, so access to functionality from there is easy.
+- The various pre-defined color constants like `peniko::Color::YELLOW` are no longer available.
+  Instead, use the CSS palette provided within `color`:  `peniko::color::palette::css::YELLOW`.
+- Similarly, parsing a color string is now provided by the `color` crate.
+- The MSRV is now 1.82 as that is the MSRV of `color`.
```
Comment (DJMcNab) [inline CHANGELOG.md:38]: MSRV bumps are not breaking changes by our policies.

File: src/brush.rs [hunk 13]
```diff
@@ -77,7 +91,11 @@ impl Brush {
 /// This is useful for methods that would like to accept brushes by reference. Defining
 /// the type as `impl<Into<BrushRef>>` allows accepting types like `&LinearGradient`
 /// directly without cloning or allocating.
-#[derive(Copy, Clone, PartialEq, Debug)]
+#[cfg_attr(
+    target_pointer_width = "32",
```
Comment (DJMcNab) [inline src/brush.rs:95]: :upside_down_face:

**Review by DJMcNab — APPROVED** [review]

File: CHANGELOG.md (hunk 12 above)
Comment (waywardmonkeys) [reply CHANGELOG.md]: Removed.

**Pull request merged**